script_parameters.yml

```diff
@@ -0,0 +1,252 @@
+# versions of the variables to prevent injection attacks.
+#
+# Links:
+# * Workflows documentation: https://civis.zendesk.com/hc/en-us/articles/115004172983-Workflows-Basics
+# * Script parameter reference: https://civis.zendesk.com/hc/en-us/articles/360001579592-Configuring-Scripts-via-the-Civis-API
+# * Mistral domain specific language reference: https://docs.openstack.org/mistral/latest/user/wf_lang_v2.html
+# * YAML syntax: https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html
+
+version: '2.0'  # you always need this key to specify version 2 of the mistral DSL
+workflow:
+  tasks:
+    sql_script:
+      action: civis.scripts.sql
+      input:
+        # SQL scripts always require the IDs of the remote host and the credential
+        # you are using to connect to it.
+        remote_host_id: 0  # get it from https://api.civisanalytics.com/databases
+        credential_id: 0   # get it from https://platform.civisanalytics.com/credentials
+        params:  # This is not an exhaustive list of param types - see the script parameter reference for more
+          - name: my_boolean
+            type: bool
+          - name: my_integer
+            type: integer
+          - name: my_float
+            type: float
+          - name: my_string
+            type: string
+          - name: my_custom_credential
+            type: credential_custom
+        arguments:
+          my_boolean: true
+          my_integer: 1234
+          my_float: 0.1234
+          my_string: this is a string
+          my_custom_credential: 0  # ID of the credential
+        sql: |
+          SELECT 1;
+
+          -- Variables enclosed in double curly brackets will be replaced with the parameter values.
+          -- my_boolean: {{my_boolean}}
+          -- my_integer: {{my_integer}}
+          -- my_float: {{my_float}}
+
+          -- To prevent SQL injection attacks, you can get escaped versions of string parameters.
+          -- {{my_string.literal}} wraps the string in single quotes, changing single quotes in the string to double quotes.
+          -- {{my_string.identifier}} is in a format that is safe to be used as a Redshift identifier.
+
+          -- my_custom_credential: {{my_custom_credential.username}} {{my_custom_credential.password}}
+      on-success:
+        - container_script
+
+    container_script:
+      action: civis.scripts.container
+      input:
+        docker_image_name: civisanalytics/datascience-python
+        docker_image_tag: latest
+        required_resources:
+          cpu: 256
+          memory: 512
+          disk_space: 1
+        params:
+          # Optionally, you can specify the parameter values in the `params` list
+          # instead of in the `arguments` hash, by using the `default` key.  But
+          # this does not work for credential parameters.
+          - name: my_boolean
+            type: bool
+            default: true
```

Comment:
Whenever I'm bothering to write workflows with script parameters fully in mistral like this, it's for the sake of being able to define arguments from the workflow level and have them propagate to multiple scripts.  This example param might create some confusion because workflow parameters don't have a `bool` type (or something there breaks, not sure what exactly, but --) you have to use a string, then parse it somewhere downstream.

Reply:
We talked offline; I'll remove the boolean examples since in a workflow they aren't really more useful than string parameters, and suggesting their use here could give people an unnecessarily confusing time.